rc/filetype/css.kak

```diff
@@ -32,28 +33,78 @@ provide-module css %[
+
+  # Units
+  # ‾‾‾‾‾
+  # generated from the URL & <code> below
+  # includes #rgb, #rrggbb, #rrggbbaa as color values {3,8}
+  # 
+  # https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Values_and_Units
+  units='% cap ch cm deg dpcm dpi dppx em ex grad Hz ic in kHz lh mm ms pc pt px Q rad rem rlh s turn vb vh vi vmax vmin vw x'
+
+  logical_ops='and not only from to'
+  keywords='!important auto inherit initial unset none'
+  media_types='all print screen speech'
+
+  # easing_re='linear|ease(-(in-out|in|out))?|step-(start|end)'
+
+  # combinators='+ > ~ ||'
+  # attribute_op='= ~= |= ^= $= *='
+
+  join() { eval set -- $1; IFS="|"; echo "$*"; }
+
+  # Selectors
+  # ‾‾‾‾‾‾‾‾‾
+  # universal: *, ns|*, *|*, |*
+  # class/id: .class, #id
+  # type: element
+  # attr: [attr=val]
+
+  # order below matters
+  printf %s "
+  add-highlighter shared/css/code/tag_selectors regex \b($(join "${html_tags}"))((:[a-z:])|[\h.#]) 1:keyword
 
-add-highlighter shared/css/declaration/base default-region group
-add-highlighter shared/css/declaration/double_string region '"' (?<!\\)(\\\\)*" fill string
-add-highlighter shared/css/declaration/single_string region "'" "'"             fill string
-add-highlighter shared/css/declaration/comment region /[*] [*]/ fill comment
+  add-highlighter shared/css/code/functional_notation regex ([a-zA-Z0-9-_]+[a-zA-Z0-9])\( 1:keyword
 
-# https://developer.mozilla.org/en-US/docs/Web/CSS/length
-add-highlighter shared/css/declaration/base/ regex (#[0-9A-Fa-f]+)|((\d*\.)?\d+(ch|cm|em|ex|mm|pc|pt|px|rem|vh|vmax|vmin|vw)) 0:value
+  add-highlighter shared/css/code/logical_operators regex (\b($(join "${logical_ops}"))\b|$(join "${keywords}")) 1:keyword 1:+i
 
-add-highlighter shared/css/declaration/base/ regex ([A-Za-z][A-Za-z0-9_-]*)\h*: 1:keyword
-add-highlighter shared/css/declaration/base/ regex :(before|after) 0:attribute
-add-highlighter shared/css/declaration/base/ regex !important 0:keyword
+  add-highlighter shared/css/code/media_types regex \b($(join "${media_types}"))\b 1:+i
 
-# element#id element.class
-# universal selector
-add-highlighter shared/css/selector/ regex         [A-Za-z][A-Za-z0-9_-]* 0:keyword
-add-highlighter shared/css/selector/ regex [*]|[#.][A-Za-z][A-Za-z0-9_-]* 0:variable
+  # (after functional notation as they may contain paranthesis)
+  add-highlighter shared/css/code/psudo regex (:{1,2})([a-z-]+) 2:attribute 2:+a
+
+  add-highlighter shared/css/code/at_rules regex @[a-z-]+ 0:function 
+
+  add-highlighter shared/css/code/css_property regex ([A-Za-z][A-Za-z0-9_-]*)\h*:\h 1:operator 1:+a
+
+  add-highlighter shared/css/code/selectors regex (\*|[*]?[.][A-Za-z][A-Za-z0-9_-]+) 1:type
+  add-highlighter shared/css/code/selectors_id regex (\*|[*]?[#][A-Za-z][A-Za-z0-9_-]+) 1:type 1:+i
```

Comment:
I feel like a simpler definition of the selectors rule might be:

```suggestion
  add-highlighter shared/css/code/selectors regex \*|[.#][A-Za-z][A-Za-z0-9_-]+ 0:type
```

That is, a `*`, or a `.` or `#` followed by alphanumerics. If somebody chains selectors together like `***.class*#id` then each selector in the list should be identified individually, and still be displayed the same.

Reply:
As other editors, I wanted to highlight an ID selector (#) with italics -- as this is somewhat important and unique.

Also, the chaining you speak of is invalid, and would not even work, so it's not a valid use case as far as I know.

I will change the `1:type` to `0:type` -- in other cases above your suggestion of using  `0` is problematic, as it will highlight items not wanted.